external/storm-hdfs/src/main/java/org/apache/storm/hdfs/trident/HdfsState.java

```diff
@@ -500,6 +500,7 @@ private void updateIndex(long txId) {
             bw.write(txnRecord.toString());
             bw.newLine();
             bw.flush();
+            out.close();       // Closing the output stream in finally block causes FileNotFoundException in Azure Data Lake Store file system (adl://)
```

Comment:
Why not just wrap the output stream in a try block instead?

```
Path tmpPath = tmpFilePath(indexFilePath.toString());
TxnRecord txnRecord = new TxnRecord(txId, options.currentFile.toString(), this.options.getCurrentOffset());
try (FSDataOutputStream out = this.options.fs.create(tmpPath, true);
       BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(out));) {
    bw.write(txnRecord.toString());
    bw.newLine();
    bw.flush();
}
```

Then you don't need to worry about the close happening before the delete, even in error cases.